client/my-sites/post-share/index.jsx

```diff
@@ -51,11 +58,28 @@ class PostShare extends Component {
 		requestConnections: PropTypes.func
 	};
 
+	static FOOTER_SECTION_SCHEDULED = 'footer-section-scheduled';
+	static FOOTER_SECTION_PUBLISHED = 'footer-section-published';
+
 	state = {
 		message: PostMetadata.publicizeMessage( this.props.post ) || this.props.post.title,
 		skipped: PostMetadata.publicizeSkipped( this.props.post ) || [],
+		footerSection: PostShare.FOOTER_SECTION_SCHEDULED
+
 	};
 
+	constructor() {
+		super( ...arguments );
+
+		this.setFooterSection = this.setFooterSection.bind( this );
```

Comment:
let's remove the `constructor` here and define the `setFooterSection` bound function in the fly using arrow functions. See my comments below.